### PR TITLE
add vector seach

### LINKED ISSUE #46: Feature: Vector search on GUI
Provide a way that user can execute vector search on a loaded partition. 

Input: 
- a CSV file contains vector
- search parameters 
  - metric_type， default
  - nprobe
- specified collection/partition
- search expr(new milvus concept)

Output:  
- rows with similarity score

### PR DESCRIPTION
**Pull request by Tumao727**
this fix #46 

* add vector search ui
vector-search

* fetch results from api

### CODE REVIEW

File: client/src/utils/Validation.ts [hunk 1]
```diff
@@ -181,6 +182,23 @@ export const checkDimension = (param: {
...
+ * function to check whether value(type: number) is equal to specified value or in valid range
+ * @param param specValue and params checkRange function needed
+ * @returns whether input is valid
+ */
+export const checkSpecValueOrRange = (param: {
+  value: number;
+  min: number;
+  max: number;
+  compareValue: number;
+}): boolean => {
```
Comment (nameczz) [inline client/src/utils/Validation.ts:195]: If i only need to check value === compareValue, do i need to pass min and max?

File: client/src/pages/seach/SearchParams.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        type: 'number',
+        value,
+        validations: [
+          {
+            rule: 'require',
+            errorText: warningTrans('required', { name: label }),
+          },
+        ],
+      };
+      if (!isSearchK && (min || max)) {
```
Comment (nameczz) [inline client/src/pages/seach/SearchParams.tsx:96]: i think you use both min and max. so it should be min && max?

File: client/src/utils/Validation.ts (hunk 1 above)
Comment (Tumao727) [reply client/src/utils/Validation.ts:195]: we already have a method to check comparison, `specValueOrRange` is for condition like `{-1} ∪ [10, 20]`, so you should pass min or max if you use it

File: client/src/i18n/en/search.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const searchTrans = {
+  firstTip: '1. Enter vector value',
+  secondTip: '2. Choose collection and field',
+  thirdTip: '3. Set search parameters',
+  vectorPlaceholder: 'Please input your vector value here, e.g. [1, 2, 3, 4]',
+  collection: 'Choose Collection',
+  field: 'Choose Field',
+  startTip: 'start your vector search',
```
Comment (shanghaikid) [inline client/src/i18n/en/search.ts:8]: 'start your vector search' => 'Start your vector search',

File: client/src/i18n/en/search.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  empty: 'no result',
```
Comment (shanghaikid) [inline client/src/i18n/en/search.ts:9]: 'no result' => 'No result'

**Review by shanghaikid — CHANGES_REQUESTED** [review]

**Review by shanghaikid — APPROVED** [review]

**Review by nameczz — APPROVED** [review]

**Pull request merged**